clients/common/cblas_interface.cpp

```diff
@@ -109,6 +110,32 @@ extern "C" {
 
     //axpy
     template<>
+    void cblas_axpy<rocblas_half>( rocblas_int n,
+                            rocblas_half alpha,
+                            rocblas_half *x, rocblas_int incx,
+                            rocblas_half *y, rocblas_int incy)
+    {
+        rocblas_int abs_incx = incx >=0 ? incx : -incx;
+        rocblas_int abs_incy = incy >=0 ? incy : -incy;
+        float alpha_float = static_cast<float>(alpha);
+        std::unique_ptr<float []> x_float(new float[n*abs_incx]());
```

Comment:
std::array would allow for array notation for x_float (and y_float), but we are already using i*abs_incx for x, so we use the same indexing on the LHS and RHS of the equation.

We could pack the temp buffer, but it may be simpler to have the same incx and incy parameters in cblas_axpy as in cblas_saxpy.